tracing/tracing.go

```diff
@@ -24,7 +24,10 @@ func Instrument(h http.Handler) http.Handler {
 			slog.Debug("header absent, generated UUID", "trace_id", traceid)
 		}
 
-		req = req.WithContext(context.WithValue(req.Context(), slog.TraceID, traceid))
+		log := slog.With("trace_id", traceid)
```

Comment:
Maybe I'm missing something, but should we use the log from the ctx, enrich it and then set it back, something like:

```go
ctx := req.Context()
log := slog.FromContext().With("trace_id", traceid)
ctx := slog.WithContext(ctx, log)
req = req.WithContext(ctx)
```

Reply:
I have a feeling that slog.WithContext doesn't read well.... because even when you read as a sentence, it seems like:

"logger (slog) with the given context"

BUt that is not what we are doing, we are creating a new context, from another context, with the logger stored inside (we are not creating a "slog with the given context".

I would expect this from this method:

```
log := slog.WithContext(ctx, log)
```

It returns a slog.Logger...with something from the context (with the context).